src/inject.ts

```diff
@@ -106,29 +106,28 @@ async function whichSearcheesMatchTorrent(
 	meta: Metafile,
 	searchees: SearcheeWithLabel[],
 ): Promise<{ matches: AllMatches; foundBlocked: boolean }> {
+	const isSimilar = (searchee: SearcheeWithLabel, meta: Metafile) =>
+		areMediaTitlesSimilar(searchee.title, meta.title) ||
+		areMediaTitlesSimilar(searchee.title, meta.name) ||
+		areMediaTitlesSimilar(searchee.name, meta.name) ||
+		areMediaTitlesSimilar(searchee.name, meta.title) ||
+		meta.files.some((metaFile) =>
+			searchee.files.some((searcheeFile) =>
+				areMediaTitlesSimilar(searcheeFile.name, metaFile.name),
+			),
+		);
 	let foundBlocked = false;
 	const matches: AllMatches = [];
 	for (const searchee of searchees) {
 		const { decision } = await assessCandidate(meta, searchee, []);
 		if (decision === Decision.BLOCKED_RELEASE) {
-			foundBlocked = true;
+			if (isSimilar(searchee, meta)) foundBlocked = true;
```

Comment:
I still think this is similar logic should be inside assessCandidate

Reply:
It would just be an unnecessary check for everything but inject. It would never catch anything that passes reverse lookup or the torznab query. It also would hurt risky matching as id search can correctly match torrents with completely different naming like the title in a different language.